File: 2laboratory4go/1.go
package main

import (
	"fmt"
	"math"
)

type Shape interface {
	Area() float64
}

type Rectangle struct {
	width float64
	height float64
}

type Circle struct {
	radius float64
}

func (r *Rectangle) Area() float64 {
	return r.height * r.width
}

func (c *Circle) Area() float64 {
	return c.radius*c.radius*math.Pi
}

func ShapeArea(s Shape) {
	fmt.Println(s.Area())
}

func main() {
	one := Rectangle{width: 9.99, height: 9.99}
	two := Circle{radius: 9.99}

	ShapeArea(&one)
	ShapeArea(&two)
}

File: pr/2laboratory2c++/1.cpp
#include <iostream>
#include <string>

using namespace std;

int add(int oneVariable, int twoVariable)
{
    return oneVariable+twoVariable;
}

double add(double oneVariable, double twoVariable)
{
    return oneVariable+twoVariable;
}

string add(string oneVariable, string twoVariable)
{
    return oneVariable+twoVariable;
}

int main()
{
    int one = 1;
    int two = 2;

    cout << add(one,two) << endl;
    return 0;
}
